File: codingBat/src/javaExample/ap1/MatchUp.java
/**************************************************************************************************

    matchUp - solved
    
    Given 2 arrays that are the same length containing strings, compare the 1st string in one array
    to the 1st string in the other array, the 2nd to the 2nd and so on. Count the number of times 
    that the 2 strings are non-empty and start with the same char.
    The strings may be any length, including 0. 
***************************************************************************************************
matchUp({"aa", "bb", "cc"}, {"aaa", "xx", "bb"}) → 1
matchUp({"aa", "bb", "cc"}, {"aaa", "b", "bb"}) → 2
matchUp({"aa", "bb", "cc"}, {"", "", "ccc"}) → 1
matchUp({"", "", "ccc"}, {"aa", "bb", "cc"}) → 1
matchUp({"", "", ""}, {"", "bb", "cc"}) → 0
matchUp({"aa", "bb", "cc"}, {"", "", ""}) → 0
matchUp({"aa", "", "ccc"}, {"", "bb", "cc"}) → 1
matchUp({"x", "y", "z"}, {"y", "z", "x"}) → 0
matchUp({"", "y", "z"}, {"", "y", "x"}) → 1
matchUp({"x", "y", "z"}, {"xx", "yyy", "zzz"}) → 3
matchUp({"x", "y", "z"}, {"xx", "yyy", ""}) → 2
matchUp({"b", "x", "y", "z"}, {"a", "xx", "yyy", "zzz"}) → 3
matchUp({"aaa", "bb", "c"}, {"aaa", "xx", "bb"}) → 1
**************************************************************************************************/
package javaExample.ap1;

public class MatchUp
{

    public static void main(String[] args)
    {
    	String manchala [] = {"aa", "bb", "cc"};
    	String shor [] = {"", "", "ccc"};
    	System.out.println(matchUp(manchala,shor));
    }
    
    public static int matchUp(String[] a, String[] b) {
        int howMuch = 0;
    	for(int i=0;i<a.length;i++){
    		if(a[i]!="" && b[i]!=""){
    			if(a[i].charAt(0)==b[i].charAt(0) ){
        			howMuch++;
        		}
    		}
    	}
    	
    	return howMuch;
    }

}

File: codingBat/python/array2/matchUp.py
#######################################################################################################################
#
#    matchUp
#
#    Given arrays nums1 and nums2 of the same length, for every element in nums1, consider the
#    corresponding element in nums2 (at the same index). Return the count of the number of times
#    that the two elements differ by 2 or less, but are not equal.
#
#######################################################################################################################
#
#   matchUp({1, 2, 3}, {2, 3, 10}) â†’ 2
#   matchUp({1, 2, 3}, {2, 3, 5}) â†’ 3
#   matchUp({1, 2, 3}, {2, 3, 3}) â†’ 2
#   matchUp({5, 3}, {5, 5}) → 1
#   matchUp({5, 3}, {4, 4}) → 2
#   matchUp({5, 3}, {3, 3}) → 1
#   matchUp({5, 3}, {2, 2}) → 1
#   matchUp({5, 3}, {1, 1}) → 1
#   matchUp({5, 3}, {0, 0}) → 0
#   matchUp({4}, {4}) → 0
#   matchUp({4}, {5}) → 1
#
#######################################################################################################################
